### ISSUE TITLE
[Seeking Guidance, v4/v5] - Automatic Placement of Annotations

### ISSUE THREAD

**Issue by cd-alexslater** [antvis/G2#4548]
Is there any automatic placement strategies built into G2 to keep annotations from overlapping with marks / other annotations? I'm evaluating whether or not we will have to write something ourselves to be able to keep the automatically annotated graphs clean.

**Comment by pearmini** [comment]
Can you give me a concrete example? I'm not sure I'm understanding it correctly. A chart that is currently plotted and the result that you expected will be very helpful.

**Comment by cd-alexslater** [comment]
@pearmini Yes - so this is what happens now if you plot multiple datamarker annotations for the same point:
![image](https://user-images.githubusercontent.com/114569661/211323853-c3360e94-3542-47e2-b4d0-91636f969c9f.png)
This is what they could look like if G2 had the ability to place them automatically:
![image](https://user-images.githubusercontent.com/114569661/211323990-a150bef2-f055-4a4f-8916-4f4ff62ebfc6.png)

Basically, I was wondering if there was any functionality for G2 to prevent overlapping of annotations without me having to manually use offsets and positioning. This would make annotations dynamic with the graph, so that you can initialize a large number of annotations, and they reposition themselves so that they don't overlap with other graph elements. 

My use case for this would be to dynamically generate graphs with multiple custom HTML annotations. It will look something like this when I'm done:
<img width="208" alt="image" src="https://user-images.githubusercontent.com/114569661/211324924-2255966f-0a8e-4aeb-a8a6-5372caaa5318.png">
